pyomo/solvers/plugins/solvers/CBCplugin.py

```diff
@@ -533,14 +539,16 @@ def process_soln_file(self, results):
                     raise RuntimeError("Unexpected line format encountered in CBC solution file - line="+line)
 
                 constraint = tokens[1]
-                constraint_ax = eval(tokens[2]) # CBC reports the constraint row times the solution vector - not the slack.
-                constraint_dual = eval(tokens[3])
-                if constraint.startswith('c_'):
+                constraint_ax = float(tokens[2]) # CBC reports the constraint row times the solution vector - not the slack.
+                constraint_dual = float(tokens[3])
+                if constraint[:2] == 'c_':
                     solution.constraint[constraint] = {"Dual" : constraint_dual}
-                elif constraint.startswith('r_l_'):
-                    range_duals.setdefault(constraint[4:],[0,0])[0] = constraint_dual
-                elif constraint.startswith('r_u_'):
-                    range_duals.setdefault(constraint[4:],[0,0])[1] = constraint_dual
+                elif constraint[:2] == 'r_':
+                    # For the range constraints, supply only the dual with the largest
+                    # magnitude (at least one should always be numerically zero)
+                    if abs(constraint_dual) > 0:
```

Comment:
I worry about this causing problems with solver tolerances.  My understanding of the previous code is that by deferring this until the end of processing, we wouldn't "accidentally" report very small duals if, say the solver returned `r_l_` of 500 and `r_u_` as 1e-8.  The code here could return 1e-8 as the dual value.  Maybe only update the dual on line 551 if the current constraint_dual value is larger?

Reply:
That's a good catch. I agree that this needs to be fixed.

Reply:
@ghackebeil @jsiirola I have pushed an additional